### ISSUE TITLE
onPushTokenReceivedCallback() not call after firebase init.

### ISSUE THREAD

**Issue by nakorndev** [EddyVerbruggen/nativescript-plugin-firebase#959]
Same as #253 that was closed. But I still got this problem to my project.

```typescript
firebase.init({
    onMessageReceivedCallback: function(message) {
        console.log("Title: " + message.title);
        console.log("Body: " + message.body);
        // if your server passed a custom property called 'foo', then do this:
        console.log("Value of 'foo': " + message.data.foo);
    },
    onPushTokenReceivedCallback: function(token) {
        console.log("Firebase push token: " + token);
    }
}).then(
    instance => {
        console.log("firebase.init done");
    },
    error => {
        console.log(`firebase.init error: ${error}`);
    }
);
```

There was only shown `firebase.init done` without error. But non for `Firebase push token: ...`. Did I missing something?

**Comment by EddyVerbruggen** [comment]
Yeah, I miss a project to reproduce this issue. Please don't share a snippet as it all depends on where that code resides and where other code was added, and what the firebase config of your project looks like. Also, which platform (iOS, Android, both).

**Comment by nakorndev** [comment]
Android.

`google-services.json` (XXXXX for sensitive data).

```json
{
  "project_info": {
    "project_number": "770428255XXX",
    "firebase_url": "https://xxxxxx.firebaseio.com",
    "project_id": "xxxxxx",
    "storage_bucket": "xxxxxx.appspot.com"
  },
  "client": [
    {
      "client_info": {
        "mobilesdk_app_id": "1:770428255XXX:android:7ef016XXXXXXXXX",
        "android_client_info": {
          "package_name": "org.nativescript.XXXXXXXXXX"
        }
      },
      "oauth_client": [
        {
          "client_id": "770428255XXX-rmpqcXXXXXXXXXXXXXX.apps.googleusercontent.com",
          "client_type": 3
        }
      ],
      "api_key": [
        {
          "current_key": "AIzaSyCXXXXXXXXXXXXXXXXXX"
        }
      ],
      "services": {
        "analytics_service": {
          "status": 1
        },
        "appinvite_service": {
          "status": 1,
          "other_platform_oauth_client": []
        },
        "ads_service": {
          "status": 2
        }
      }
    }
  ],
  "configuration_version": "1"
}
```

**Comment by nakorndev** [comment]
Sorry. I forget this.

`firebase.nativescript.json`

```json
{
    "external_push_client_only": false,
    "using_ios": true,
    "using_android": true,
    "firestore": false,
    "realtimedb": false,
    "authentication": false,
    "remote_config": false,
    "messaging": true,
    "crashlytics": false,
    "crash_reporting": false,
    "storage": false,
    "functions": false,
    "facebook_auth": false,
    "google_auth": false,
    "admob": false,
    "invites": false,
    "dynamic_links": false,
    "ml_kit": false
}
```

**Comment by EddyVerbruggen** [comment]
That's not a project repo. Closing this as I can't investigate the issue this way. Feel free to open when you have a repo to share.

**Comment by nakorndev** [comment]
@EddyVerbruggen Seem I found what caused `onPushTokenReceivedCallback()`

After I changed emulator from **Visual Studio Emulator for Android** to **Android Emulator (Android Studio)** this function is called and working perfectly.

Maybe lack some kind of service from google or something from emulator. It should be implement catch error for this? It seems on #253 they are using some emulator that doesn't work with some emulator like me.
